base/sim/FairGenericStack.h

```diff
@@ -111,12 +112,25 @@ class FairGenericStack : public TVirtualMCStack
      *@param en          new energy      of the particle
      **/
     virtual void          FastSimMoveParticleTo(Double_t xx, Double_t yy, Double_t zz, Double_t tt,
-                                                Double_t px, Double_t py, Double_t pz, Double_t en) {}
+                                                Double_t px, Double_t py, Double_t pz, Double_t en);
     /** Fast simulation function to stop original particle. **/
-    virtual void          FastSimStopParticle  () {}
-    /** Allow FairFastSim the retrieval of moved particle position **/
-    virtual Int_t         FastSimGetMovedIndex () { return -1; }
-    virtual void          FastSimClearMovedIndex () {}
+    virtual void          FastSimStopParticle  ();
+    /** Fast simulation function to generate secondaries.
+     *@param xx,yy,zz    position    of the particle
+     *@param tt          proper time of the particle
+     *@param px,py,pz    momentum    of the particle
+     *@param en          energy      of the particle
+     **/
+    virtual void          FastSimPushSecondary(Int_t parentID, Int_t pdgCode,
+                                               Double_t xx, Double_t yy, Double_t zz, Double_t tt,
+                                               Double_t px, Double_t py, Double_t pz, Double_t en,
+                                               Double_t polx, Double_t poly, Double_t polz, TMCProcess proc,
+                                               Double_t weight, Int_t is);
+    /** Allow FairFastSim the retrieval of moved particle position, p1 and p2 to get secondaries **/
+    virtual Int_t FastSimGetMovedIndex (Int_t& p1, Int_t& p2) { p1 = fFSFirstSecondary; p2 = fFSNofSecondaries; return fFSMovedIndex; }
```

Comment:
```suggestion
    virtual std::tuple<Int_t, Int_t, Int_t> FastSimGetMovedIndex() { return std::make_tuple(fFSMovedIndex, fFSFirstSecondary, fFSNofSecondaries); }
```

See [CppCoreGuidelines F.21](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rf-out-multi).

Calling example:

```cpp
// C++11
int firstSecondary(0);
int nofSecondaries(0);
int movedParticleIndex(0);
std::tie(movedParticleIndex, firstSecondary, nofSecondaries) = ((FairGenericStack*)(gMC->GetStack()))->FastSimGetMovedIndex();

// C++17
auto [movedParticleIndex, firstSecondary, nofSecondaries] = ((FairGenericStack*)(gMC->GetStack()))->FastSimGetMovedIndex();
```

It would also need `#include <tuple>`.